packages/ui/src/helpers/accountSettings/validator.ts

```diff
@@ -0,0 +1,168 @@
+import { Amplify } from 'aws-amplify';
+import { hasSpecialChars } from '../authenticator';
+import {
+  ValidatorSpec,
+  PasswordSettings,
+  PasswordRequirement,
+  ValidationMode,
+  InputEventType,
+} from '../../types';
+
+// gets password requirement from Amplify.configure data
+export const getPasswordRequirement = (): PasswordRequirement => {
+  // need to cast to any because `Amplify.configure()` isn't typed properly
+  const config = Amplify.configure() as any;
+  const passwordSettings =
+    config?.aws_cognito_password_protection_settings as PasswordSettings;
+
+  if (!passwordSettings) {
+    return null;
+  }
+
+  const {
+    passwordPolicyCharacters: characterPolicy = [],
+    passwordPolicyMinLength: minLength,
+  } = passwordSettings;
+
+  return {
+    minLength: minLength ? minLength : undefined,
```

Comment:
This ternary looks it might be extraneous unless we're worried abt falsy values for `minLength` that aren't `undefined`?

Reply:
+1, simplified in [2a1d469](https://github.com/aws-amplify/amplify-ui/pull/2900/commits/2a1d4697f25d2ac57e036941880f21df26fc4248)